src/Components/Forms/src/EditContextModelBindingContextExtensions.cs

```diff
@@ -0,0 +1,104 @@
+    /// <returns></returns>
+    public static IDisposable EnableBindingContextExtensions(this EditContext context, ModelBindingContext bindingContext)
+    {
+        ArgumentNullException.ThrowIfNull(context, nameof(context));
+        ArgumentNullException.ThrowIfNull(bindingContext, nameof(bindingContext));
+
+        context.Properties[_key] = bindingContext;
+
+        return new BindingContextEventSubscriptions(context, bindingContext);
+    }
+
+    /// <summary>
+    /// Gets the attempted value for the specified field name.
+    /// </summary>
+    /// <param name="context">The <see cref="EditContext"/>.</param>
+    /// <param name="fieldName">The field name.</param>
+    /// <returns></returns>
+    public static string? GetAttemptedValue(this EditContext context, string fieldName)
+    {
+        ArgumentNullException.ThrowIfNull(context, nameof(context));
+        ArgumentNullException.ThrowIfNull(fieldName, nameof(fieldName));
+
+        if (context.Properties.TryGetValue(_key, out var result) && result is ModelBindingContext bindingContext)
+        {
+            return bindingContext.GetAttemptedValue(fieldName);
+        }
+
+        return null;
+    }
+
+    private sealed class BindingContextEventSubscriptions : IDisposable
+    {
+        private readonly EditContext _editContext;
+        private readonly ModelBindingContext _bindingContext;
+        private readonly ValidationMessageStore _messages;
+
+        public BindingContextEventSubscriptions(EditContext editContext, ModelBindingContext serviceProvider)
+        {
+            _editContext = editContext;
+            _bindingContext = serviceProvider;
+            _messages = new ValidationMessageStore(_editContext);
+
+            _editContext.OnValidationRequested += OnValidationRequested;
+        }
+
+        private void OnValidationRequested(object? sender, ValidationRequestedEventArgs e)
+        {
+            foreach (var (key, errors) in _bindingContext.GetAllErrors())
+            {
+                FieldIdentifier fieldIdentifier;
+                var lastDotIndex = key.LastIndexOf('.');
+                if (lastDotIndex >= 0)
+                {
+                    fieldIdentifier = new FieldIdentifier(_editContext.Model, key.Substring(lastDotIndex + 1), key);
+                }
+                else if(key != "")
+                {
+                    fieldIdentifier = new FieldIdentifier(_editContext.Model, key, key);
+                }
+                else
+                {
+                    fieldIdentifier = new FieldIdentifier(_editContext.Model, fieldName: string.Empty);
+                }
+
+                foreach (var error in errors)
+                {
+                    // TODO: We need to support localizing the error message.
+                    _messages.Add(fieldIdentifier, error.ToString(CultureInfo.InvariantCulture));
+                }
+            }
+
+            _editContext.NotifyValidationStateChanged();
```

Comment:
Ideally, only call this if you have changed the validation state, as it will trigger a re-render.

Reply:
Yeah, I've changed the logic here to capture the errors the first time we are called, and do nothing on subsequent calls.